sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -185,8 +184,9 @@ protected ServiceSpec getServiceSpec() {
                 .setPrincipal(serviceSpec.getPrincipal())
                 .setFailoverTimeout(failoverTimeoutSec)
                 .setUser(userString)
-                .setCheckpoint(true)
-                .setRole(serviceSpec.getRole());
+                .setCheckpoint(true);
+
+        getRoles(serviceSpec).forEach(role -> fwkInfoBuilder.addRoles(role));
```

Comment:
looks like this could be `fwkInfoBuilder.addRoles(serviceSpec.getRole());` (could even be in the above chained construction of `fwkInfoBuilder`)